tests/rptest/tests/transactions_test.py

```diff
@@ -0,0 +1,222 @@
+        extra_rp_conf = {
+            "enable_idempotence": True,
+            "enable_transactions": True,
+            "transaction_coordinator_replication": 3,
+            "id_allocator_replication": 3,
+            "enable_leader_balancer": False,
+            "enable_auto_rebalance_on_node_add": False
+        }
+
+        super(TransactionsTest, self).__init__(test_context=test_context,
+                                               extra_rp_conf=extra_rp_conf)
+
+        self.input_t = self.topics[0]
+        self.output_t = self.topics[1]
+        self.max_records = 100
+
+    def generate_data(self, topic, num_records):
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+        })
+
+        for i in range(num_records):
+            producer.produce(topic.name, str(i), str(i), 0)
+
+        producer.flush()
+
+    @cluster(num_nodes=3)
+    def simple_test(self):
+        self.generate_data(self.input_t, self.max_records)
+
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': '0',
+            'transaction.timeout.ms': 10000,
+        })
+
+        consumer = ck.Consumer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'group.id': "test",
+            'auto.offset.reset': 'earliest',
+            'enable.auto.commit': False,
+        })
+
+        producer.init_transactions()
+
+        consumer.subscribe([self.input_t])
+
+        num_consumed_records = 0
+        consume_for_one_it = 10
+
+        while num_consumed_records != self.max_records:
+            records = consumer.consume(consume_for_one_it)
+            assert (len(records) != 0)
+
+            producer.begin_transaction()
+
+            for record in records:
+                assert (record.error() == None)
+                producer.produce(self.output_t.name, record.value(),
+                                 record.key(), 0)
+
+            producer.send_offsets_to_transaction(
+                consumer.position(consumer.assignment()),
+                consumer.consumer_group_metadata())
+
+            producer.commit_transaction()
+
+            num_consumed_records += consume_for_one_it
+
+        consumer = ck.Consumer({
+            'group.id': "testtest",
+            'bootstrap.servers': self.redpanda.brokers(),
+            'auto.offset.reset': 'earliest',
+        })
+        consumer.subscribe([self.output_t])
+
+        final_consume_cnt = 0
+        while True:
+            record = consumer.poll(consume_for_one_it)
+            if record is None:
+                break
+
+            expected = bytes(str(final_consume_cnt), 'UTF-8')
+            assert (record.key() == expected)
+            assert (record.value() == expected)
+            final_consume_cnt += 1
+
+        assert (final_consume_cnt == self.max_records)
+
+    @cluster(num_nodes=3)
+    def rejoin_member_test(self):
```

Comment:
Think we are missing some coverage for consumers offsets_fetch_request request with/without RequireStable. We should probably check that it blocks until no conflicting transaction is in progress?

Reply:
We did it inside chaos test

Reply:
It is branch in Denis's fork :D 
I think we should have pr for it

Reply:
since you guys were talking about the chaos PR, I've created [it](https://github.com/redpanda-data/chaos/pull/15)